thanos/charts/templates/ruler.yaml

```diff
@@ -0,0 +1,32 @@
+{{/* 
+SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Greenhouse contributors
+SPDX-License-Identifier: Apache-2.0
+*/}}
+
+{{- if and (.Capabilities.APIVersions.Has "monitoring.coreos.com/v1") .Values.thanos.ruler.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ThanosRuler
+metadata:
+  name: {{ include "release.name" . }} # ressources name = thanos-ruler-<name>
+  labels:
+    {{- include "plugin.labels" . | nindent 4 }}
+    {{- include "thanos.labels" . | nindent 4 }}
+    {{- if .Values.thanos.ruler.labels }}
+    {{ toYaml .Values.thanos.ruler.labels | nindent 4 }}
+    {{- end }}
+spec:
+  image: "{{ .Values.thanos.image.repository }}:{{ .Values.thanos.image.tag | default .Chart.AppVersion }}"
+  ruleSelector:
+    matchLabels:
+      thanos-ruler: {{ .Values.thanos.ruler.matchLabel | default .Release.Name }} # Label for fetching rules
+  queryEndpoints:
+    - {{ include "release.name" . }}-query:10902
+  alertmanagersConfig:
+    key: alertManagerConfig.yaml # Key name for alert manager config yaml
+    name: {{ include "release.name" . }}-alertmanager-config # Secret name containing alert manager config
```

Comment:
do we have this config around?
I think this is something which needs to be added to communicate with the alertmanager. as this is a configmap it should be part of this PR. The necessary values need to be provided when enabling the ruler. Document it in the thanos readme what and how to specify it, and then it needs to go in the plugindefinition.

Reply:
if configured, prometheus should already have the certs to communicate with alertmanager. ideally I would reuse those but in the mean time I can make it same setup as current prod
https://github.com/sapcc/helm-charts/blob/master/common/thanos/templates/ruler/alertmanager-config.yaml

Reply:
fair point. the ruler and query also could run in a cluster without prometheus, so then we would need this on top.

Reply:
i would have to look into this more, but if alertmanager is deployed we should be able to fetch TLS secrets and generate config for thanos. let's discuss this offline, i will add the configmap to templates for now which will require some user input when deploying the plugin